docs/application/native/guides/text-input/voice-control-manager.md

```diff
@@ -0,0 +1,1131 @@
+
+  ```c
+  void start_recording(bool exclusive_command_option)
+  {
+      int ret;
+      ret = vc_mgr_start(exclusive_command_option);
+      if (VC_ERROR_NONE != ret)
+          /* Error handling */
+  }
+  ```
+
+- To stop recording, use `vc_mgr_stop()`. The recording stops and the voice control service state is changed to `VC_SERVICE_STATE_PROCESSING`. When the recognition command result is processed, the `vc_result_cb` callback is invoked and the voice control service state changes back to `VC_SERVICE_STATE_READY`. `vc_mgr_stop()` must be called when the voice control service is in the `VC_SERVICE_STATE_RECORDING` state:
+
+  ```c
+  void stop_recording()
+  {
+      int ret;
+      ret = vc_mgr_stop();
+      if (VC_ERROR_NONE != ret)
+          /* Error handling */
+  }
+  ```
+
+- To cancel the recording, use `vc_mgr_cancel()`. This function must be called when the voice control service is in the `VC_SERVICE_STATE_RECORDING` or `VC_SERVICE_STATE_PROCESSING` state:
+
+  ```c
+  void cancel_recording()
+  {
+      int ret;
+      ret = vc_mgr_cancel();
+      if (VC_ERROR_NONE != ret)
+          /* Error handling */
+  }
+  ```
+
+<a name="send_requests"></a>
+## Send requests
+
+You can send requests using the voice control manager:
+
+- To send the event information to the voice control engine for activating specific action, use `vc_mgr_do_action()`. This function must be called when the voice control manager is in the `VC_STATE_READY` state:
+
+  ```c
+  void send_do_action(vc_send_event_type type, char* send_event)
+  {
+      int ret;
+      ret = vc_mgr_do_action(type, send_event);
+      if (VC_ERROR_NONE != ret)
+          /* Error handling */
+  }
+  ```
+
+- To send the event and request message to a specific voice control engine, use `vc_mgr_send_specific_engine_request()`. This function must be called when the voice control manager is in the `VC_STATE_READY` state:
+
+  ```c
+  void send_specific_engine_request(const char* engine_app_id, const char* event, const char* request)
+  {
+      int ret;
+      ret = vc_mgr_send_specific_engine_request(engine_app_id, event, request);
+      if (VC_ERROR_NONE != ret)
+          /* Error handling */
+  }
+  ```
+
+<a name="info"></a>
+## Retrieve voice control information
+
+To get information about the current states, service states, current and supported languages:
+
+- Get the current voice control manager state using `vc_mgr_get_state()`. The voice control manager state changes according to the function calls:
+
+  ```c
+  void get_state()
+  {
+      vc_state_e current_state;
+      int ret;
+      ret = vc_mgr_get_state(&current_state);
+      if (VC_ERROR_NONE != ret)
+          /* Error handling */
+  }
+  ```
+
+- Get the current voice control manager service state using `vc_mgr_get_service_state()`. If the application uses continuous recognition, the voice control service state can be changed from `VC_SERVICE_STATE_PROCESSING` directly to `VC_SERVICE_STATE_RECORDING`:
```

Comment:
Checked this line in raw format, the spacing is fine. It is rendering with spaces due to code style.

![image](https://user-images.githubusercontent.com/43839737/111575680-ec2b2880-87d4-11eb-8659-a1e69a684819.png)
![image](https://user-images.githubusercontent.com/43839737/111575698-fbaa7180-87d4-11eb-9744-a88a12c22f26.png)